src/Security/Authentication/Core/src/AuthenticationHandler.cs

```diff
@@ -16,6 +17,8 @@ public abstract class AuthenticationHandler<TOptions> : IAuthenticationHandler w
         private Task<AuthenticateResult>? _authenticateTask;
 
         public AuthenticationScheme Scheme { get; private set; } = default!;
+
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
```

Comment:
Does the attribute do anything when the property isn't settable?

Reply:
This is a good question. It probably doesn't need to be on here but the linker complains about the backing field for the property. It's possible this attribute should automagically flow to the compiler generated backing field to avoid that.

The other thing is that we have a new constraint so this shouldn't be needed at all... Likely a linker issue.

Reply:
This is weird - the attribute on this property doesn't make any sense - it should only be on properties of type `System.Type` or `System.String`. Linker currently doesn't warn about this (I think it should - https://github.com/mono/linker/issues/1420) - but in reality it will simply ignore the attribute in this case.

I don't know what the original warning was about the backing field you mention. If it has to do with `TOptions` then the attribute should be on the generic parameter. (with the linker fix it might not be needed due to the new constraint anyway).

Reply:
I'll remove these on the properties

Reply:
> I'll remove these on the properties

It's just this one place that is an issue, right?

> with the linker fix it might not be needed due to the new constraint anyway

That fix has been checked in (https://github.com/mono/linker/pull/1414). @davidfowl - can you try the latest mono/linker and see if the warning just goes away here?